### PR TITLE
Fixes for Analytics hit response logic

### PR DESCRIPTION
**Pull request by addb**
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: code/analytics/src/main/java/com/adobe/marketing/mobile/analytics/internal/AnalyticsHitProcessor.kt [hunk 1]
```diff
@@ -147,8 +145,10 @@ internal class AnalyticsHitProcessor(
                             AnalyticsConstants.EventDataKeys.Analytics.CONTENT_TYPE_HEADER
                         )
                     )
+                    val responseString = StreamUtils.readAsString(connection.inputStream)
                     val eventData: Map<String, Any?> = mapOf(
-                        AnalyticsConstants.EventDataKeys.Analytics.ANALYTICS_SERVER_RESPONSE to httpHeaders,
+                        AnalyticsConstants.EventDataKeys.Analytics.ANALYTICS_SERVER_RESPONSE to responseString,
+                        AnalyticsConstants.EventDataKeys.Analytics.HEADERS_RESPONSE to httpHeaders,
                         AnalyticsConstants.EventDataKeys.Analytics.HIT_HOST to url,
                         AnalyticsConstants.EventDataKeys.Analytics.HIT_URL to payload,
```
Comment (emdobrin) [inline code/analytics/src/main/java/com/adobe/marketing/mobile/analytics/internal/AnalyticsHitProcessor.kt:153]: can you also update host and url to the correct values?

File: code/analytics/src/main/java/com/adobe/marketing/mobile/analytics/internal/AnalyticsHitProcessor.kt [hunk 2]
```diff
@@ -16,16 +16,14 @@ import com.adobe.marketing.mobile.Event
...
 import com.adobe.marketing.mobile.ExtensionApi
-import com.adobe.marketing.mobile.services.DataEntity
-import com.adobe.marketing.mobile.services.HitProcessing
-import com.adobe.marketing.mobile.services.HitProcessingResult
-import com.adobe.marketing.mobile.services.HttpMethod
-import com.adobe.marketing.mobile.services.Log
-import com.adobe.marketing.mobile.services.NetworkRequest
-import com.adobe.marketing.mobile.services.Networking
-import com.adobe.marketing.mobile.services.ServiceProvider
+import com.adobe.marketing.mobile.services.*
```
Comment (emdobrin) [inline code/analytics/src/main/java/com/adobe/marketing/mobile/analytics/internal/AnalyticsHitProcessor.kt:19]: let's expand this to only the used classes

File: code/analytics/src/main/java/com/adobe/marketing/mobile/analytics/internal/AnalyticsHitProcessor.kt (hunk 1 above)
Comment (addb) [reply code/analytics/src/main/java/com/adobe/marketing/mobile/analytics/internal/AnalyticsHitProcessor.kt:153]: I verified this with our internal project.

File: code/analytics/src/main/java/com/adobe/marketing/mobile/analytics/internal/AnalyticsHitProcessor.kt [hunk 3]
```diff
@@ -147,8 +145,10 @@ internal class AnalyticsHitProcessor(
                             AnalyticsConstants.EventDataKeys.Analytics.CONTENT_TYPE_HEADER
                         )
                     )
+                    val responseString = StreamUtils.readAsString(connection.inputStream)
```
Comment (emdobrin) [inline code/analytics/src/main/java/com/adobe/marketing/mobile/analytics/internal/AnalyticsHitProcessor.kt:148]: We should close the inputStream once it is no longer needed

File: code/analytics/src/main/java/com/adobe/marketing/mobile/analytics/internal/AnalyticsHitProcessor.kt (hunk 1 above)
Comment (kevinlind) [reply code/analytics/src/main/java/com/adobe/marketing/mobile/analytics/internal/AnalyticsHitProcessor.kt:156]: Yes, confusing, but the iOS implementation is the same: https://github.com/adobe/aepsdk-analytics-ios/blob/main/AEPAnalytics/Sources/AnalyticsHitProcessor.swift#L144

File: code/analytics/src/main/java/com/adobe/marketing/mobile/analytics/internal/AnalyticsHitProcessor.kt [hunk 4]
```diff
@@ -178,25 +176,29 @@ internal class AnalyticsHitProcessor(
...
                     Log.warning(
                         AnalyticsConstants.LOG_TAG,
                         CLASS_NAME,
                         "processHit - Dropping Analytics hit, request with url $url failed with error and unrecoverable status code ${connection.responseCode}"
                     )
-                    processingResult.complete(true)
+                    doneProcessingResult = true
                 }
             }
+            if (doneProcessingResult) {
```
Comment (kevinlind) [inline code/analytics/src/main/java/com/adobe/marketing/mobile/analytics/internal/AnalyticsHitProcessor.kt:198]: The connection needs to be closed either way here, so can you remove this check?  If the request is retried, then a new connection is created.

**Review by kevinlind — APPROVED** [review]

**Review by emdobrin — APPROVED** [review]

**Pull request merged**